common/src/main/java/io/netty/util/internal/ThreadCleaner.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2017 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.util.internal;
+
+import java.lang.ref.ReferenceQueue;
+import java.lang.ref.WeakReference;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * Allows a way to register some {@link Runnable} that will executed once there are no references to the {@link Thread}
+ * anymore. This typically happens once the {@link Thread} dies / completes.
+ */
+public final class ThreadCleaner {
+
+    // This will hold a reference to the ThreadCleanerReference which will be removed once we called cleanup()
+    private static final Set<ThreadCleanerReference> LIVE_SET = new ConcurrentSet<ThreadCleanerReference>();
+    private static final ReferenceQueue<Object> REFERENCE_QUEUE = new ReferenceQueue<Object>();
+    private static final AtomicBoolean CLEANER_RUNNING = new AtomicBoolean(false);
```

Comment:
nit: caps are usually for constants.

Reply:
I think CAPS are for things that are static final so I think this is fine. Sure the content can be changed but this is ok IMHO.